src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,185 @@
+                } catch(ArrayIndexOutOfBoundsException e) {
+                    return new ErrorCommand("☹ OOPS!!! The description of a todo cannot be empty.");
+                }
+            case "deadline" :
+                try {
+                    String taskName = input.split(" /by ", 2)[0].split(" ", 2)[1];
+
+                    // ArrayList containing possible formats of Date and Time.
+                    List<String> formatStrings = Arrays.asList(
+                            "yyyy-MM-dd HH:mm",
+                            "dd/MM/yyyy HH:mm",
+                            "MM-dd-yyyy HH:mm"
+                            // Add other formats here
+                    );
+                    String dateInput = input.split(" /by ", 2)[1];
+                    String dueDate = null;
+
+                    // Try parsing with different formats
+                    for (String formatString : formatStrings) {
+                        try {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
+                            LocalDateTime dateTime = LocalDateTime.parse(dateInput, formatter);
+                            dueDate = dateTime.format(formatter);
+                            break;  // Stop at the first successful parse
+                        } catch (DateTimeParseException e) {
+                            // Ignore the exception and try the next format
+                        }
+                    }
+
+                    // if the input fits none of the format, throw DateTimeParseException
+                    if (dueDate == null) {
+                        throw new DateTimeParseException("Invalid date/time format", dateInput, 0);
+                    }
+
+                    // add item into list
+                    Deadline task = new Deadline(taskName, dueDate);
+                    return new AddCommand(task);
+                } catch ( ArrayIndexOutOfBoundsException e) {
+                    if (input.split(" ").length == 1) {
+                        return new ErrorCommand("☹ OOPS!!! The description of a deadline cannot be empty");
+                    } else {
+                        return new ErrorCommand("☹ OOPS!!! The description of a deadline does not have \"/by\" specified");
+                    }
+                } catch (DateTimeParseException e) {
+                    return new ErrorCommand("☹ OOPS!!! The format of deadline is wrong, please write it in the correct format");
+                }
+            case "event" :
+                try {
+
+                    String taskName = input.split(" /from | /to ", 3)[0].split(" ", 2)[1];
+                    String startDateInput = input.split(" /from | /to ", 3)[1];
+                    String dueDateInput = input.split(" /from | /to ", 3)[2];
+
+                    // ArrayList containing possible formats of Date and Time.
+                    List<String> formatStrings = Arrays.asList(
```

Comment:
May be the variable can be better named? For example, name it as stringFormat to make it sound like a thing, instead of a action verb.